Breeding-Insight/bi-web
# [BI-1356] Ordinal Nominal Switching Errors

# Description
**Story:** [ISSUE: New/Edit Term - Switching between scale classes, specifically ordinal and nominal, not working properly](https://breedinginsight.atlassian.net/browse/BI-1356)

Refactoring of Ordinal/Nominal display to get rid of anti-pattern, fix problem with ordinal/nominal category fields not displaying on first selection, and make code more robust.

- Functionality for modifying category "data" prop was moved from child component CategoryTraitForm.vue to parent component BaseTraitForm in order to avoid anti-pattern problems.
- Categories are now being passed in as categories (which is set to this.trait.scale.categories) due to the nested this.trait.scale.categories prop messing with reactivity
- Switching between ordinal/nominal now effectively hides the "value" fields rather than switching around item.value and item.label when changing scale class and maintaining a history. 
  - The category fields are consistent between ordinal and nominal display, with item.label being passed in as the category for both ValueRow.vue (used for nominal) and LabelValueRow.vue (used for ordinal). item.value is passed in as the value in LabelValueRow.vue, and is effectively "hidden" when switching from ordinal to nominal selection.
  - On save, if the scale class is nominal, the value and label are switched around in order to properly save to the database, and on load they are switched to maintain this more robust display.

# Dependencies
bi-api/release/0.5

# Testing

- Create new ontology term, select nominal, make sure one category displays and categories can be added and removed switch to ordinal and see that there are at minimum two fields and information is preserved
- Create new ontology term, select ordinal, make sure two categories display and categories can be added and removed switch to ordinal and see that there are at minimum two fields and information is preserved
- Save new ordinal term successfully
- Save new nominal term successfully
- Open existing nominal ontology term, check categories display properly, click edit, check display properly, save, check save properly
- Open existing ordinal ontology term, check categories display properly, click edit, check display properly, save, check save properly

# Checklist:

- [X] I have run TAF: [_\<link to TAF run>_](https://github.com/Breeding-Insight/taf/actions/runs/1959345494)
